src/js/compat.ts

```diff
@@ -268,6 +269,9 @@ export async function calculateDirname(): Promise<string> {
 
     // FIXME: We would like to use import.meta.url here,
     // but mocha seems to mess with compiling typescript files to ES6.
+    if (IN_BUN) {
+      return nodeUrl.fileURLToPath(`file://${nodePath.dirname(fileName)}`);
```

Comment:
Does this not work in node? If we could always prepend with `file://` we could drop `IN_BUN`.

Reply:
Looks like the nodejs and bun are not the same here.

```
node --experimental-wasm-stack-switching makesnap.mjs
node:internal/url:796
    this.#updateContext(bindingUrl.parse(input, base));
                                   ^

TypeError: Invalid URL
    at new URL (node:internal/url:796:36)
    at Module.fileURLToPath (node:internal/url:1457:12)
    at calculateDirname (file:///workspaces/pyodide/dist/pyodide.mjs:3:3203)
    at async loadPyodide (file:///workspaces/pyodide/dist/pyodide.mjs:3:8740)
    at async file:///workspaces/pyodide/dist/makesnap.mjs:7:12 {
  code: 'ERR_INVALID_URL',
  input: 'file://file:///workspaces/pyodide/dist'
}

Node.js v20.12.2
make: *** [Makefile:241: dist/snapshot.bin] Error 1
```

Reply:
Hmm, this is strange. Node also throws an error if there is no `file://` prefix:

```node
> url.fileURLToPath('/asdf/asdf')
Uncaught TypeError [ERR_INVALID_URL]: Invalid URL
    at __node_internal_captureLargerStackTrace (node:internal/errors:484:5)
    at new NodeError (node:internal/errors:393:5)
    at URL.onParseError (node:internal/url:564:9)
    at new URL (node:internal/url:644:5)
    at Object.fileURLToPath (node:internal/url:1484:12) {
  input: '/asdf/asdf',
  code: 'ERR_INVALID_URL'
}
```

So, I wonder what the difference is between the two. Could you please check the values of

```
let fileName = ErrorStackParser.parse(err)[0].fileName!;
```

and
```
nodePath.dirname(fileName)
```

?

I wonder if `ErrorStackParser` is not working correctly with Bun's stack trace.

Reply:
I added a `console.log` here:

```js
    console.log({ fileName, dirname: nodePath.dirname(fileName) })
```

During `make` it prints:

```
...
[2024-07-11 11:15:07] done building packages...
cp src/templates/console.html dist/console.html
sed -i -e 's#{{ PYODIDE_BASE_URL }}#./#g' dist/console.html
cd dist && node --experimental-wasm-stack-switching makesnap.mjs
{
  fileName: 'file:///workspaces/pyodide/dist/pyodide.mjs',
  dirname: 'file:///workspaces/pyodide/dist'
}
```